### PR TITLE
[WIP] hugepage requests should only be allowed by G or Bu pods

### PR DESCRIPTION
**Pull request by tanshanshan**
**What this PR does / why we need it**:
When pod QoS is  BestEffort, no need to calculate hugepages. ref #44817

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by xiangpengzhao** [comment]
@tanshanshan  please check and fix the test failures :)

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go [hunk 1]
```diff
@@ -581,10 +581,13 @@ func PodFitsResources(pod *v1.Pod, meta algorithm.PredicateMetadata, nodeInfo *s
 			predicateFails = append(predicateFails, NewInsufficientResourceError(rName, podRequest.ExtendedResources[rName], nodeInfo.RequestedResource().ExtendedResources[rName], allocatable.ExtendedResources[rName]))
 		}
 	}
-
-	for rName, rQuant := range podRequest.HugePages {
-		if allocatable.HugePages[rName] < rQuant+nodeInfo.RequestedResource().HugePages[rName] {
-			predicateFails = append(predicateFails, NewInsufficientResourceError(rName, podRequest.HugePages[rName], nodeInfo.RequestedResource().HugePages[rName], allocatable.HugePages[rName]))
+	var podBestEffort bool
+	podBestEffort = isPodBestEffort(pod)
+	if !podBestEffort {
```
Comment (xiangpengzhao) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: How about changing this line to `if !isPodBestEffort(pod) {` and remove the above two lines?

**Comment by tanshanshan** [comment]
@davidopp @derekwaynecarr , PTAL

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go [hunk 2]
```diff
@@ -578,8 +578,12 @@ func PodFitsResources(pod *v1.Pod, meta algorithm.PredicateMetadata, nodeInfo *s
 	}
 
 	for rName, rQuant := range podRequest.ScalarResources {
+		if isPodBestEffort(pod) && v1helper.IsHugePageResourceName(rName) {
```
Comment (derekwaynecarr) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: no scheduler changes are needed.

**Review by derekwaynecarr — CHANGES_REQUESTED** [review]
my intention per https://github.com/kubernetes/kubernetes/issues/53672 is to prevent a pod from making a hugepage request if it does not make a corresponding cpu or memory request.  

the code change we actually need is in the apiserver validation code path.

**Pull request closed without merging**